lib/pure/asyncmacro.nim

```diff
@@ -447,8 +325,28 @@ macro multisync*(prc: untyped): untyped =
   result = newStmtList()
   result.add(asyncSingleProc(asyncPrc))
   result.add(sync)
-
-proc await*[T](x: T) =
-  ## The 'await' keyword is also defined here for technical
-  ## reasons. (Generic symbol lookup prepass.)
-  {.error: "Await only available within .async".}
+  # echo result.repr
+
+# overload for type which is not Future[T], based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
+template await*(f: untyped): untyped =
```

Comment:
AFAIK `untyped` here will make

```
await:
  if true: echo 24
  echo 123
```

Possible. Can we change to `typed`?

Reply:
it won't as you probably see, because it would lead to an error : await expected Future[T] etc

Reply:
ah, you mean outside of `async`?